src/rules.jl

```diff
@@ -0,0 +1,64 @@
+## Square
+SQUARE_PAIRS = (
```

Comment:
I'd put these variables in a `let` block (or do `for (a,b) in (...)`), otherwise you're bloating the compiled cache with stuff which isn't needed at runtime

Reply:
Thanks for the review! I'll update the code later.